graph/edge/impl/ThingEdgeImpl.java

```diff
@@ -436,17 +499,15 @@ public void isInferred(boolean isInferred) {
          *
          * This operation can only be performed once, and thus protected by
          * {@code isDelete} atomic boolean. The delete operation involves
-         * removing this edge from the {@code from.outs()} and {@code to.ins()}
-         * edge collections in case it is cached. Then, delete both directions
-         * of this edge from the graph storage.
+         * removing this edge from the graph storage and notifying the from/to vertices of the modification.
          */
         @Override
         public void delete() {
             if (deleted.compareAndSet(false, true)) {
-                fromWritable().outs().remove(this);
-                toWritable().ins().remove(this);
```

Comment:
this is important and subtle - we no longer cache the persisted edge on read in the `from` and `to`. Therefore, we no longer need to remove the edge when we delete it. However, we still need to set the vertices as modified so they are able to commit & also tracked correctly for consistency violations.